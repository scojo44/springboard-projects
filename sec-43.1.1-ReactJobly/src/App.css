#root {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
}
main {
  display: flex;
  flex: 1;
  max-height: calc(100vh - 3rem);
}
