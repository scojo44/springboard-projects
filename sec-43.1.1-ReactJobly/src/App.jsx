import React from 'react'
import NavBar from './NavBar'
import RoutesList from './RoutesList'
import './App.css'

function App() {
  return (
    <>
      <NavBar />
      <main>
        <RoutesList />
      </main>
    </>
  )
}

export default App
