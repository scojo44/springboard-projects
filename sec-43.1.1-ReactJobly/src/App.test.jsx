import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import App from './App';

describe('App Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
