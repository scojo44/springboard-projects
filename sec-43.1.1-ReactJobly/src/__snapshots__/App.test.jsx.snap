// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="NavBar"
  >
    <a
      href="/"
    >
      Jobly
    </a>
    <a
      class=""
      href="/signup"
    >
      Sign Up
    </a>
    <a
      class=""
      href="/login"
    >
      Log In
    </a>
  </nav>
  <aside
    class="Alert"
  />
  <main>
    <section
      class="Home"
    >
      <h2>
        Welcome, guest!
      </h2>
    </section>
  </main>
</DocumentFragment>
`;
